### PR TITLE
UGWP v0 v1 combined

### LINKED ISSUE #296: Error in regression test compare netcdf files
## Description
On Orion, @jiandewang notes regression tests passing when he knows the results are different from the baseline.  Logfile indicates the alternate comparison method (compare_ncfile.py) is being used.

### To Reproduce:
I have not been able to reproduce; in my run, the same regression tests failed with compare_ncfile detecting differences in netcdf files

**output logs**
Jiande's:
```
baseline dir = /work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20201118/INTEL/cpld_control_ccpp
working dir  = /work/noaa/stmp/jiwang/stmp/jiwang/FV3_RT/rt_133263/cpld_control_prod
Checking test 001 cpld_control results ....
 Comparing phyf024.tile1.nc ............ALT CHECK......OK
 Comparing dynf024.tile1.nc ............ALT CHECK......OK
 Comparing RESTART/coupler.res .........OK
 Comparing RESTART/fv_core.res.nc .........OK
 Comparing RESTART/fv_core.res.tile1.nc ............ALT CHECK......OK
```
Mine:
```
baseline dir = /work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20201118/INTEL/cpld_control_ccpp
working dir  = /work/noaa/stmp/jminsuk/stmp/jminsuk/FV3_RT/rt_222345/cpld_control_prod
Checking test 001 cpld_control results ....
 Comparing phyf024.tile1.nc ............ALT CHECK......NOT OK
 Comparing dynf024.tile1.nc ............ALT CHECK......NOT OK
 Comparing RESTART/coupler.res .........OK
 Comparing RESTART/fv_core.res.nc .........OK
 Comparing RESTART/fv_core.res.tile1.nc ............ALT CHECK......NOT OK
```

### LINKED ISSUE #347: The new CCCPP suite FV3_GFS_v16b_ugwpv1 and regression tests for GFSv16b-127L/C96 configuration
## Description
The new suite represents the upgrade of uGWP (V1), CU-CIRES. It is capable to reproduce the realistic FV3GFS climate, including equatorial dynamics (QBO and SAO, 20-60 km) and annual cycles in the extra-tropics and polar regions in the upper atmosphere (20-80km). It can be executed and tested with the gsldrag sub-grid scale orography physics of gsl/develop for the UFS-R2O tests in 2021.

## Solution
The introduction and commit of new suite FV3_GFS_v16b_ugwpv1 will require  3PRs in:
1) ufs-weather-model ; 2) fv3atm; 3) cccp-physics; for performance of the warm-start

For performance of the cold-start it requires update of ufs-weather-model/FMS/sat_vapor_pres/sat_vapor_pres.F90
(tmaxc=160) to make the cold-start from "unbalanced" analyses during diagnostics of RH-values in the upper layers (mesosphere) for atmos_4xdaily.tile1.nc on the first time step.

Therefore the optional 4PD in  ufs-weather-model/FMS maybe considered for the top-lid (127L) UFS.

## Alternatives
N/A

## Related to
Repositories that will be touched:   ufs-weather-model ,  fv3atm,   ccpp-physics
The user's feature branches on 
https://github.com/ValeryYudin-NOAA/ufs-weather-model               branch  ugwpv1_gsldrag
                                   /fv3atm                                                              branch  ugwpv1_gsldrag                 
                                   /ccpp-physics                                                   branch  ugwpv1_gsldrag  

It requires also the work on the upgrade of the 127-level INPUT and RESTART data inside:
 /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20201220/FV3_input_data_127
that can be replaced by user's input directory for FV3-127L configuration
/scratch1/NCEPDEV/swpc/Valery.Yudin/noscrub/CCPP_GEOS/FV3_input_data

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR
- combines the changes in PRs #360 and #382
- adds three regression tests `fv3_ccpp_gfsv16_ugwpv1 `, `fv3_ccpp_gfsv16_ugwpv1_warmstart` and `fv3_ccpp_gfsv16_ugwpv1_debug`
- contains updates and bugfixes for `nc_compare.py` and the CI tests from @MinsukJi-NOAA 
- update Python3 environment on jet.intel, gaea.intel, cheyenne.{intel,gnu}
- turn off (again) test `fv3_ccpp_decomp` on jet.intel, this test didn't work in the past, but recently it "passed", because the error checking with `nc_compare.py` failed silently and we didn't notice it

### Issue(s) addressed

Fixes #347.
Fixes https://github.com/ufs-community/ufs-weather-model/issues/296.

## Testing

Initial testing (before the FMS update) on Hera with Intel showed that all of the existing regression tests are b4b reproducible, and that the three newly introduced run to completion.

Further testing (after the FMS update) on Hera with Intel and GNU showed that all of the existing regression tests are b4b reproducible, and that the three newly introduced run to completion.

Final regression tests passed on hera.gnu, hera.intel, orion.intel, cheyenne.gnu, cheyenne.intel, wcoss_cray, wcoss_dell_p3, jet.intel, gaea.intel

## Dependencies

https://github.com/NCAR/ccpp-physics/pull/561
https://github.com/NOAA-EMC/fv3atm/pull/241
https://github.com/ufs-community/ufs-weather-model/pull/396

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
The wamstart test is not set up for or intended to be a restart test, right?

**Comment by climbfuji** [comment]
> The wamstart test is not set up for or intended to be a restart test, right?

No, it's just warmstart files as input.

**Comment by climbfuji** [comment]
@MinsukJi-NOAA CI tests passed after I pulled in your changes, thanks!

File: tests/compare_ncfile.py [hunk 1]
```diff
@@ -4,12 +4,20 @@
...
 
   for varname in nc1.variables.keys():
-    diff = nc2[varname][:]-nc1[varname][:]
-    if (np.abs(diff)).max() != 0:
-      print(varname,"is different")
-      sys.exit(1)
+    # First check if each variable has the same dimension
+    if np.shape(nc1[varname][:])!=np.shape(nc2[varname][:]):
+      print(varname,"dimension is different")
+      sys.exit(2)
```
Comment (BrianCurtis-NOAA) [inline tests/compare_ncfile.py:16]: raise

File: tests/compare_ncfile.py [hunk 2]
```diff
@@ -4,12 +4,20 @@
...
+    # If dimension is the same, compare data
+    else:
+      diff = nc2[varname][:]-nc1[varname][:]
+
+      if (np.abs(diff)).max() != 0:
+        print(varname,"is different")
+        sys.exit(2)
```
Comment (BrianCurtis-NOAA) [inline tests/compare_ncfile.py:23]: raise

File: tests/compare_ncfile.py [hunk 3]
```diff
@@ -4,12 +4,20 @@
 from netCDF4 import Dataset
 
 with Dataset(sys.argv[1]) as nc1, Dataset(sys.argv[2]) as nc2:
+  # Check if the list of variables are the same
   if nc1.variables.keys()!=nc2.variables.keys():
     print("Variables are different")
-    sys.exit(1)
+    sys.exit(2)
```
Comment (BrianCurtis-NOAA) [inline tests/compare_ncfile.py:10]: should use raise instead of sys.exit()

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by BrianCurtis-NOAA — CHANGES_REQUESTED** [review]
try to use `raise` instead of `sys.exit`. There's some default exception types that cover some of these like `TypeError` or `KeyError`. Simplest is just `raise Exception('explanation here')`

**Comment by climbfuji** [comment]
> try to use `raise` instead of `sys.exit`. There's some default exception types that cover some of these like `TypeError` or `KeyError`. Simplest is just `raise Exception('explanation here')`

@MinsukJi-NOAA FYI

I would prefer to make this change in a future PR, because it will require rerunning the regression tests.

**Comment by BrianCurtis-NOAA** [comment]
> > try to use `raise` instead of `sys.exit`. There's some default exception types that cover some of these like `TypeError` or `KeyError`. Simplest is just `raise Exception('explanation here')`
> 
> @MinsukJi-NOAA FYI
> 
> I would prefer to make this change in a future PR, because it will require rerunning the regression tests.

This is fine with me.

**Comment by MinsukJi-NOAA** [comment]
> > try to use `raise` instead of `sys.exit`. There's some default exception types that cover some of these like `TypeError` or `KeyError`. Simplest is just `raise Exception('explanation here')`
> 
> @MinsukJi-NOAA FYI
> 
> I would prefer to make this change in a future PR, because it will require rerunning the regression tests.

OK. I will note this for future improvements.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
> > > try to use `raise` instead of `sys.exit`. There's some default exception types that cover some of these like `TypeError` or `KeyError`. Simplest is just `raise Exception('explanation here')`
> > 
> > 
> > @MinsukJi-NOAA FYI
> > I would prefer to make this change in a future PR, because it will require rerunning the regression tests.
> 
> This is fine with me.

Thanks, Brian and Minsuk.

**Comment by climbfuji** [comment]
PR is ready for merging, CI tests have just been kicked off.

**Review by junwang-noaa — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA CI tests passed, please merge when you are ready. Thanks!

**Pull request merged**